### PR TITLE
Adds support for Domains in CSV source

### PR DESCRIPTION
**Pull request by pedro93**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/graph/client.py [hunk 1]
```diff
@@ -196,6 +197,13 @@ def get_glossary_terms(self, entity_urn: str) -> Optional[GlossaryTermsClass]:
             aspect_type=GlossaryTermsClass,
         )
 
+    def get_domain(self, entity_urn: str) -> Optional[DomainsClass]:
```
Comment (aditya-radhakrishnan) [inline metadata-ingestion/src/datahub/ingestion/graph/client.py:200]: Thank you!

File: metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py [hunk 2]
```diff
@@ -576,6 +636,13 @@ def get_workunits(self) -> Iterable[MetadataWorkUnit]:
                     row, is_resource_row
                 )
 
+                # Does this make sense?
```
Comment (aditya-radhakrishnan) [inline metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py:639]: This makes sense!

File: metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py [hunk 3]
```diff
@@ -551,6 +600,17 @@ def maybe_extract_owners(
         ]
         return owners
 
+    def maybe_extract_domain(
+        self, row: Dict[str, str], is_resource_row: bool
+    ) -> List[DomainsClass]:
```
Comment (aditya-radhakrishnan) [inline metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py:605]: I'm a little confused why this is a list. Can we just return a DomainsClass?

Comment (aditya-radhakrishnan) [reply metadata-ingestion/src/datahub/ingestion/source/csv_enricher.py:605]: Ah I see we don't use this method. Can you just get rid of it then?

**Review by aditya-radhakrishnan — APPROVED** [review]
Overall looks good. If we can just delete that maybe_extract_domain method I think we're good to go!

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**